### PR TITLE
Allow CSMA-CA to wait for an idle timeslot before CCA.

### PR DESCRIPTION
**Pull request by hubertmis**
This is an optional workaround for MAC layers that does not handle
correctly multiple CHANNEL_BUSY reports in a row.

### CODE REVIEW

**Comment by hubertmis** [comment]
@wbober @tomchy 
Have a look at this new configuration parameter. It's going to be included in incoming release.

File: src/nrf_802154_config.h [hunk 1]
```diff
@@ -422,6 +422,22 @@ extern "C" {
 #define NRF_802154_CSMA_CA_MAX_CSMA_BACKOFFS 4
 #endif
 
+/**
+ * @def NRF_802154_CSMA_CA_WAIT_FOR_TIMESLOT
+ *
+ * If the CSMA-CA algorithm waits for an available timeslot before it performs the CCA procedure.
```
Comment (b-gent) [inline src/nrf_802154_config.h]:
``` suggestion
 * Indicates whether the CSMA-CA algorithm waits for an available timeslot before it performs the CCA procedure.
```

File: src/nrf_802154_config.h [hunk 2]
```diff
@@ -422,6 +422,22 @@ extern "C" {
...
+ *
+ * @note When this option is enabled, the CSMA-CA procedure may be synchronized with timeslots of
+ *       other protocols. It decreases robustness of the CSMA-CA procedure and is not recommended.
+ *       This option may be used as a workaround to problems with higher layers that does not handle
```
Comment (b-gent) [inline src/nrf_802154_config.h]:
``` suggestion
 *       This option may be used as a workaround to problems with higher layers that do not correctly handle
```

File: src/nrf_802154_config.h [hunk 3]
```diff
@@ -422,6 +422,22 @@ extern "C" {
...
+ *       correctly multiple CHANNEL_BUSY errors in a row. This potentially problematic situation may
```
Comment (b-gent) [inline src/nrf_802154_config.h]:
``` suggestion
 *       multiple CHANNEL_BUSY errors in a row. This potentially problematic situation may
```

**Pull request merged**